core/silkworm/common/endian.hpp

```diff
@@ -72,26 +72,28 @@ as compiler intrinsics to swap bytes in 16-bit, 32-bit, and 64-bit integers resp
 namespace silkworm::endian {
 
 #if SILKWORM_BYTE_ORDER == SILKWORM_LITTLE_ENDIAN
-struct le {
-    static inline uint16_t uint16(uint16_t value) noexcept { return value; }
-    static inline uint32_t uint32(uint32_t value) noexcept { return value; }
-    static inline uint64_t uint64(uint64_t value) noexcept { return value; }
+struct LE {
+    static inline uint16_t load(uint16_t value) noexcept { return value; }
+    static inline uint32_t load(uint32_t value) noexcept { return value; }
+    static inline uint64_t load(uint64_t value) noexcept { return value; }
+    static inline intx::uint256 load(const intx::uint256& value) noexcept { return value; }
 };
-struct be {
-    static inline uint16_t uint16(uint16_t value) noexcept { return SILKWORM_BSWAP16(value); }
-    static inline uint32_t uint32(uint32_t value) noexcept { return SILKWORM_BSWAP32(value); }
-    static inline uint64_t uint64(uint64_t value) noexcept { return SILKWORM_BSWAP64(value); }
+struct BE {
+    static inline uint16_t load(uint16_t value) noexcept { return SILKWORM_BSWAP16(value); }
+    static inline uint32_t load(uint32_t value) noexcept { return SILKWORM_BSWAP32(value); }
+    static inline uint64_t load(uint64_t value) noexcept { return SILKWORM_BSWAP64(value); }
+    static inline intx::uint256 load(const intx::uint256& value) noexcept { return intx::bswap(value); }
 };
 #elif SILKWORM_BYTE_ORDER == SILKWORM_BIG_ENDIAN
-struct le {
-    static inline uint16_t uint16(uint16_t value) noexcept { return SILKWORM_BSWAP16(value); }
-    static inline uint32_t uint32(uint32_t value) noexcept { return SILKWORM_BSWAP32(value); }
-    static inline uint64_t uint64(uint64_t value) noexcept { return SILKWORM_BSWAP64(value); }
+struct LE {
+    static inline uint16_t load(uint16_t value) noexcept { return SILKWORM_BSWAP16(value); }
+    static inline uint32_t load(uint32_t value) noexcept { return SILKWORM_BSWAP32(value); }
+    static inline uint64_t load(uint64_t value) noexcept { return SILKWORM_BSWAP64(value); }
```

Comment:
no intx::uint256 version here?

Reply:
No because I suspect that intx::uint256 might have some issues on big-endian platforms. In any case, currently there are some places in the Silkworm codebase where we assume that the platform is little-endian. IMHO supporting big-endian platforms is not a priority.